test/e2e/internal/utils/match/requests.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package match
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/onsi/gomega/gbytes"
+)
+
+// requestKeywordMatcher provides keyword matching for request content.
+// It looks into the debug output of an operation and the match will pass
+// if all the keywords exist in every sent request.
+type requestKeywordMatcher []string
+
+// NewRequestKeywordMatcher returns a request keyword matcher.
+func NewRequestKeywordMatcher(kw []string) requestKeywordMatcher {
+	return requestKeywordMatcher(kw)
+}
+
+// Match matches got with wanted keywords.
+func (r requestKeywordMatcher) Match(got *gbytes.Buffer) {
+	var missed []string
+
+	raw := string(got.Contents())
+	reqs := processRequests(raw)
+	for _, req := range reqs {
+		for _, w := range r {
+			if !strings.Contains(req, w) {
+				missed = append(missed, w)
+			}
+		}
+	}
+
+	if len(missed) != 0 {
+		fmt.Printf("Keywords missed: %v\n", missed)
+		panic("failed to match all keywords")
+	}
+}
+
+// processRequests parses raw debug output to a string slice
+// containing each request.
+func processRequests(debugOutput string) []string {
+	reqs := strings.Split(debugOutput, "request url:")
+	reqs = reqs[1:]
+	return reqs
+}
```

Comment:
might also need a function like `func getHeaders(requestLogs []string) []string {` to extract header out.

Reply:
Added this function.